api/src/main/java/org/openmrs/Concept.java

```diff
@@ -710,12 +710,16 @@ public ConceptName getName(Locale locale, boolean exact) {
 		
 		ConceptName exactName = getNameInLocale(locale);
 		
-		if (exactName != null || exact) {
+		if (exactName != null) {
 			return exactName;
 		}
 		
-		//just get any name
-		return getName();
+		if (exact == false) {

```

Comment:
@djazayeri Sorry for that Style of coding.However the change you advised is already updated in master https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/Concept.java#L714